test/limitOrderBook/limitOrderBook.fillLimitOrder.triggerPrice.ts

```diff
@@ -0,0 +1,457 @@
+        })
+
+        it("fill stop limit order: B2Q (short) exact input", async () => {
+            // a limit order to short exact 0.1 ETH for a minimum of $290 at limit price $2900
+            // fill price is guaranteed to be >= limit price
+            const triggerPrice = parseEther("3000")
+            const stopLimitOrder = {
+                orderType: fixture.orderTypeStopLimitOrder,
+                salt: 1,
+                trader: trader.address,
+                baseToken: baseToken.address,
+                isBaseToQuote: true,
+                isExactInput: true,
+                amount: parseEther("0.1").toString(),
+                oppositeAmountBound: parseEther("290").toString(), // lower bound of output quote
+                deadline: ethers.constants.MaxUint256.toString(),
+                sqrtPriceLimitX96: 0,
+                referralCode: ethers.constants.HashZero,
+                reduceOnly: false,
+                roundIdWhenCreated: computeRoundId(1, 1),
+                triggerPrice: triggerPrice.toString(),
+            }
+
+            const signature = await getSignature(fixture, stopLimitOrder, trader)
+            const orderHash = await getOrderHash(fixture, stopLimitOrder)
+
+            const roundIdWhenTriggered = computeRoundId(1, 3) // 2900
+            const triggeredPrice = await limitOrderBook.getPriceByRoundId(baseToken.address, roundIdWhenTriggered)
+            expect(triggeredPrice).to.be.lte(triggerPrice)
+
+            await expect(limitOrderBook.connect(keeper).fillLimitOrder(stopLimitOrder, signature, roundIdWhenTriggered))
+                .to.emit(limitOrderBook, "LimitOrderFilled")
+                .withArgs(
+                    trader.address,
+                    baseToken.address,
+                    orderHash,
+                    keeper.address,
+                    fixture.rewardAmount,
+                    parseEther("-0.1"), // exchangedPositionSize
+                    parseEther("295.998435782542603038"), // exchangedPositionNotional
+                    parseEther("2.959984357825426031"), // fee
+                )
+
+            // trigger price is not matched
+            const stopLimitOrder2 = {
+                ...stopLimitOrder,
+                salt: 2,
+            }
+            const signature2 = await getSignature(fixture, stopLimitOrder2, trader)
+
+            const roundIdWhenTriggered2 = computeRoundId(2, 1) // 3100
+            await expect(
+                limitOrderBook.connect(keeper).fillLimitOrder(stopLimitOrder2, signature2, roundIdWhenTriggered2),
+            ).to.revertedWith("LOB_SSLOTPNM")
+        })
+    })
+
+    describe("take profit limit order", async () => {
+        let currentTime: number
+
+        beforeEach(async () => {
+            currentTime = (await waffle.provider.getBlock("latest")).timestamp
+
+            await setRoundData(mockedBaseAggregator, computeRoundId(1, 1), "2700", currentTime)
```

Comment:
Can merge this with the above `beforeEach` block.
Apart from this, tests look nice to me!